integration-tests/contract-security-tickets-join.spec.ts

```diff
@@ -0,0 +1,198 @@
+        const opJoinContract = await opJoin.contract();
+        expect(await opJoinContract.storage()).toBeTruthy();
+
+        const opCaller = await Tezos.contract.originate({
+          code: `   {   parameter address;
+                        storage unit;
+                        code
+                          {
+                            CAR; # drop storage, but leave input on stack
+                            CONTRACT (option (ticket string));
+                            IF_NONE { FAIL } {};
+                            PUSH mutez 0;
+                            PUSH nat 1;
+                            PUSH string "test";
+                            TICKET;
+                            SOME;
+                            TRANSFER_TOKENS;
+                            NIL operation;
+                            SWAP;
+                            CONS;
+                            UNIT;
+                            SWAP;
+                            PAIR;
+                          };
+                        }`,
+          init: 'Unit',
+        });
+
+        await opCaller.confirmation();
+        expect(opCaller.hash).toBeDefined();
+        expect(opCaller.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opCallerContract = await opCaller.contract();
+        expect(await opCallerContract.storage()).toBeTruthy();
+        const opSend = await opCallerContract.methods.default(`(Pair ${opCallerContract.address} \"test\" 1)`).send();
+        await opSend.confirmation()
+      } catch (error: any) {
+        expect(error.message).toContain('michelson_v1.invalid_primitive');
+      }
+      done();
+    });
+
+    mondaynet('Verify creating ticket is not possible with two different, but similar cty', async (done) => {
+      try {
+        const op = await Tezos.contract.originate({
+          code: `   { parameter unit;
+                      storage unit;
+                      code
+                        {
+                          DROP; # drop input and storage
+                          PUSH nat 4;
+                          PUSH string "test";
+                          TICKET;
+                          PUSH nat 1;
+                          PUSH string "tes";
+                          TICKET;
+                          PAIR;
+                          JOIN_TICKETS;
+                          IF_NONE { FAIL } {};
+                          DROP;
+                          UNIT;
+                          NIL operation;
+                          PAIR;
+                        };
+                      }`,
+          init: 'Unit',
+        });
+
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opContract = await op.contract();
+        expect(await opContract.storage()).toBeTruthy();
+        const opSend = await opContract.methods.default(`(Pair ${opContract.address} \"test\" 1)`).send();
```

Comment:
wrong parameter `(Pair ${opContract.address} \"test\" 1)`
should be unit

Reply:
the purpose of the test is to verify that tickets cannot be created by cheating with bogus parameters